packages/windows/src/RPA/Windows/keywords/window.py

```diff
@@ -367,14 +376,64 @@ def close_current_window(self) -> bool:
 
             ${status}=  Close Current Window
         """
-        if not self.ctx.window:
-            self.ctx.logger.warning("There is no active window")
+        window = self.window
+        if window is None:
+            self.logger.warning("There is no active window!")
+            self.ctx.window_element = None
             return False
-        pid = self.ctx.window.item.ProcessId
-        name = self.ctx.window.item.Name
-        self.ctx.logger.info(
-            'Closing window with Name:"%s", ProcessId: %s' % (name, pid)
-        )
+
+        pid = window.item.ProcessId
+        self.logger.info("Closing window with name: %s (PID: %d)", window.name, pid)
         os.kill(pid, signal.SIGTERM)
```

Comment:
This is more me wondering about the Python API, but it's funky that we use kill with SIGTERM here since a) there is no SIGTERM signal on Windows and b) this library is Windows only. I know that os.kill straight up kills the process as a compatibility thing, but feels... odd.

Reply:
I guess you wanted to say there's no `SIGKILL` on Windows (while on *nix there are both):
```python
>>> import signal
>>> signal.SIGKILL  # happens on Windows only (exists on Mac/Linux)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'signal' has no attribute 'SIGKILL'
>>> signal.SIGTERM
<Signals.SIGTERM: 15>
```

To my understanding `os.kill` is more like a badly named function for interrupting a process with a mandatory signal parameter.
> The SIGTERM signal is a generic signal used to cause program termination. Unlike SIGKILL , this signal can be blocked, handled, and ignored. It is the normal way to politely ask a program to terminate. The shell command kill generates SIGTERM by default.

https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html#:~:text=The%20SIGTERM%20signal%20is%20a,kill%20generates%20SIGTERM%20by%20default.

Reply:
I was referring to the fact that Windows doesn't have POSIX style signals at all. It has events for interrupts and methods to terminate processes, but not signals. There are just various places where they're automatically handled for compatibility reasons. For os.kill [the documentation](https://docs.python.org/3/library/os.html#os.kill) says the following:

> Windows: The `signal.CTRL_C_EVENT` and `signal.CTRL_BREAK_EVENT` signals are special signals which can only be sent to console processes which share a common console window, e.g., some subprocesses. Any other value for sig will cause the process to be unconditionally killed by the TerminateProcess API, and the exit code will be set to sig. The Windows version of `kill()` additionally takes process handles to be killed.

Not that it really matters here. That function call does what we want, it's just a bit odd.